pkg/cyclic/replication.go

```diff
@@ -0,0 +1,134 @@
+// Copyright 2020 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package cyclic contains scaffolds for implementing cyclic replication
+// among mutliple TiDB clusters/MySQL. It uses a mark table to identify and
+// filter duplicate DMLs.
+// CDC needs to watch DMLs to mark tables and ignore all DDLs to mark tables.
+//
+// Note for now, mark tables must be create manually.
+package cyclic
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/pingcap/ticdc/cdc/model"
+	"github.com/pingcap/ticdc/pkg/filter"
+)
+
+const (
+	// SchemaName is the name of schema where all mark tables are created
+	SchemaName string = "tidb_cdc"
+	tableName  string = "repl_mark"
+
+	// CyclicReplicaIDCol is the name of replica ID in mark tables
+	CyclicReplicaIDCol string = "replica_id"
+
+	// OptCyclicConfig is the key that adds to changefeed options
+	// automatically is cyclic replication is on.
+	OptCyclicConfig string = "_cyclic_relax_sql_mode"
+)
+
+// RelaxSQLMode returns relaxed SQL mode, "STRICT_TRANS_TABLES" is removed.
+func RelaxSQLMode(oldMode string) string {
+	toRemove := "STRICT_TRANS_TABLES"
+
+	if !strings.Contains(oldMode, toRemove) {
+		return oldMode
+	}
+
+	// concatenated by "," like: mode1,mode2
+	newMode := strings.Replace(oldMode, toRemove+",", "", -1)
+	newMode = strings.Replace(newMode, ","+toRemove, "", -1)
+	newMode = strings.Replace(newMode, toRemove, "", -1)
+	return newMode
+}
+
+// Cyclic ...
```

Comment:
```suggestion
// Cyclic ...
```
the comment of `Cyclic` is missing